docs/docs/concepts/05-building-blocks/01-signals.md

```diff
@@ -0,0 +1,88 @@
+---
+sidebar_position: 1
+---
+
+# Signals
+
+In the AICA framework, signals are designed to exchange continuous data between components and controllers in a running
+application. They are an abstraction of ROS 2 topics.
+
+ROS 2 topics are messages sent from publishers to subscribers under a specific namespace (the message "topic")
+using some predefined message format (the message "type"). As a result, topics enable ROS nodes to communicate in many
+different network topologies and message formats.
+
+Often, the message topic and message type is hardcoded within the implementation of a ROS node. This can make it
+difficult to rearrange the network topology of ROS nodes without modifying and recompiling the node implementation
+itself.
+
+Signals then are ROS 2 publishers and subscribers with dynamically assigned topics and standardized message types.
+This makes it easy to reconfigure the signal connections between different components and controllers in the application
+graph without modifying or recompiling any source code. By using standard message types, the signal compatibility
+between components and controllers is also simplified.
+
+<!-- TODO: not sure if this is too much to put here, since it's more relevant to the programming reference
+Additionally, ROS 2 messages are data packets, not data objects. Parsing data from a message, manipulating it and
+writing it back into a message can involve a fair amount of boiler-plate code.
+
+When developing an AICA component, signals are automatically converted into the corresponding data object.
+-->
+
+:::note
+
+The distinction between signals and topics is also made for a different reason; signals are treated as a continuous,
+periodic data exchange (though of course, computationally the messages still happen at discrete intervals). This is in
+contrast to AICA events, which use topics to trigger discrete logical behavior.
+
+:::
+
+## Signal types
+
+The following standard message types are provided for signals.
+
+- Boolean (true / false)
+- Integer (whole numbers)
+- Double (floating point numbers)
+- Vector (array of floating point numbers)
+- String (plain text)
+
+### State messages
+
+In robot control applications, the _state_ of a robot or other objects is highly important.
```

Comment:
nitpick: Again, there is nothing incorrect with the way you wrote it but here, I got the feeling that the state message type was not highlighted enough. When I read it, I saw the list of standard messages, then you see the state types from state representation but the `make it easy ... state message types` is in my opinion not prominent enough and doesn't highlight the fact that this is (probably) the main topic type and a big simplification.